Lukeshu/tidy

LukeShu opened the pull request:
## Description
A few sentences describing the overall goals of the pull request's commits.

## Related Issues
List related issues.

## Testing
A few sentences describing what testing you've done, e.g., manual tests, automated tests, deployed in production, etc.

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [ ] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [ ] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [ ] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

ddymko reviewed (changes requested):
@LukeShu these changes look good I am just wondering why are we doing snake casing on the imports?

Shouldn't those be camelCase? Any IDE or VSCode with the Go extensions will flag those.

LukeShu commented:
> @LukeShu these changes look good I am just wondering why are we doing snake casing on the imports?
> 
> Shouldn't those be camelCase? Any IDE or VSCode with the Go extensions will flag those.

Both underscores and camel case are equally invalid; if VSCode complains about one and not the other then that's a problem with VSCode.

https://go.dev/blog/package-names

> Good package names are short and clear. They are lower case, with no `under_scores` or `mixedCaps`.

> When renaming an imported package, the local name should follow the same guidelines as package names (lower case, no `under_scores` or `mixedCaps`).

... but protobuf naming conventions and ECP's conventions are very much at odds with that.  Shame on protobuf and on the envoyproxy/go-control-plane maintainers. But we need to do something about it; and this is the clearest least-bad thing I've come up with.  If someone else comes up with a different naming convention that conveys the same info but doesn't use underscores or mixed caps, I'm all ears.

LukeShu commented:
Rebased up to `master` to resolve conflicts. Only real change was that I split adding `goimports` to `.golangci.yml` and running `make format/go` in to two commits.
